python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,706 @@
+                                # node type ntype1 are located.
+                            }
+                        feature2: {
+                            ...
+                            },
+                        ...
+                        },
+                    ntype2: {
+                        ...
+                        },
+                    ...
+
+                }
+            * ``edge_data``[optional], a dictionary which contains the metadata
+                    for the features associated with edge types from the graph.
+                    This dictionary is identical to the ``node_data``
+                    dicionary, but it will describe the metadata which will
+                    be used to locate files for edge features for edge types
+                    from the input graph.
+
+    A valid metadata.json file should adhere to the following rules.
+        * ``graph_name`` should be a valid string with non-zero length,
+        * ``num_nodes_per_type`` should be a list of positive non-zero integers
+                and the length of list should be more than 0,
+        * ``num_edges_per_type`` should be a list of positive non-zero integers
+                and the length of list should be more than 0,
+        * ``node_type`` should ba list of valid non-None strings of positive
+                length, length of this list should be equal to the
+                ``num_nodes_per_type`` list,
+        * ``edge_type`` should ba list of valid non-None strings of positive
+                length, length of this list should be equal to the
+                ``num_edges_per_type`` list,
+        * ``edges`` should have the same no. of keys are the no. of edge types
+                present in the ``edge_type`` list,
+        * ``node_data``, if present, can contain keys whose count is less than
+                or equal to the node types present in the ``node_type`` list,
+                Each node type in this dictionary is expected to non-zero
+                features,
+        * ``edge_data``, if present, follows the same rules as described for
+                ``node_data`` in the earlier bullet point.
+
+    Examples:
+    ---------
+    Creating and initializing the an instance of this class during the pipelines
+    initialization process.
+    >>> base_dir = "/home/ubuntu/graph-dataset"
+    >>> file_path = "metadata.json"
+    >>> metadata_obj = Metadata(file_name, base_dir)
+
+    Once the object of this class is created it can be stored for retrieving any
+    information from the Metadata file when needed.
+    >>> graph_ntypes = metadata_obj.ntypes
+    >>> print(graph_ntypes)
+    >>> ["ntype1", "ntype2"]
+
+    In the above code snipped ``ntypes`` property is invoked to get a list of
+    node types present in the input graph, whose return value is a list of
+    strings. The above code snippet assumes that the input graph has two
+    node types namely ``ntype1`` and ``ntype2``.
+    """
+
```

Comment:
Please provide descriptions of input args.

Reply:
Done.